app/containers/ActionEdit/index.js

```diff
@@ -34,40 +35,90 @@ import {
 import messages from './messages';
 import { save } from './actions';
 
-export class ActionEdit extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
+export class ActionEdit extends React.Component { // eslint-disable-line react/prefer-stateless-function
 
   componentWillMount() {
     this.props.loadEntitiesIfNeeded();
-
-    if (this.props.action && this.props.actionsReady) {
-      this.props.populateForm('actionEdit.form.action', fromJS(this.props.action.attributes));
+    if (this.props.action && this.props.dataReady) {
+      this.props.populateForm('actionEdit.form.data', this.getInitialFormData());
     }
   }
 
   componentWillReceiveProps(nextProps) {
-    if (nextProps.action && nextProps.actionsReady && !this.props.actionsReady) {
-      this.props.populateForm('actionEdit.form.action', fromJS(nextProps.action.attributes));
+    // console.log('componentWillReceiveProps', nextProps, this.props)
+    // repopulate if new data becomes ready
+    if (nextProps.action && nextProps.dataReady && !this.props.dataReady) {
+      this.props.populateForm('actionEdit.form.data', this.getInitialFormData(nextProps));
+    }
+    // reload entities if invalidated
+    if (this.props.action && !nextProps.action && !nextProps.dataReady) {
+      this.props.loadEntitiesIfNeeded();
     }
```

Comment:
TODO: re-consider conditions for triggering reload. current logic: if we had an action before and not anymore, we better make sure entities are loaded again